src/app/helpers/helperMethods.ts

```diff
@@ -32,4 +38,69 @@ export namespace helperMethods {
         }
         return false;
     };
+
+    export async function downloadProjectTemplate(projectFolder: string, projectRepo: string, projectBranch: string): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                await request(`${projectRepo}/archive/${projectBranch}.zip`)
+                .pipe(fs.createWriteStream(zipFile))
+                .on('error', function () {
+                    throw new Error("unable to download project zip file")
+                })
+                .on('close', async (err) => {
+                  await unzipProjectTemplate(projectFolder);
+                  resolve();
+            });
+        } catch (err) {
+            reject(err);
+        }
+    });
+    }
+
+    async function unzipProjectTemplate(projectFolder: string): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            const zipFile = 'project.zip';
+            const readStream = fs.createReadStream(`${projectFolder}/${zipFile}`);
+            readStream.pipe(unzip.Extract({ path: projectFolder }))
+                .on('error', function () {
+                    throw new Error("unable to unzip project zip file")
```

Comment:
Please format error messages nicely.  "Unable to unzip the project zip file."
Use a semicolon at the end of the line.